### ISSUE TITLE
Incorrect documentation in ``gumble_softmax`` function.

### ISSUE THREAD

**Issue by miltonllera** [pytorch/pytorch#77053]
### 📚 The doc issue

[Gumbel-Softmax documentation](https://pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html) states that the ``logits`` argument should be unnormalized. However, this is not how the formula is defined in the [original article](https://arxiv.org/abs/1611.01144), but apparently it's how it is coded in the accompanying notebook. The issue was referenced in this [PyTorch forum post](https://discuss.pytorch.org/t/gumbel-softmax-function-logits/22797).

The problem is that unnormalized log-probabilities may be to large w.r.t. the values introduced by the random samples from the Gumbel distribution, which defeats the purpose of the layer. This in turn leads to poor training performance.

### Suggest a potential alternative/fix

Two possible fixes: 
1. Change the documentation to state that the ``logits`` should be normalized.
2. Normalize the logits inside the function using ``log_softmax``.

**Comment by Boltzmachine** [comment]
I think this post is true and the implementation needs to be fixed.
see torch forum [[Gumbel_softmax function logits?](https://discuss.pytorch.org/t/gumbel-softmax-function-logits/22797)](https://discuss.pytorch.org/t/gumbel-softmax-function-logits/22797) and [Function `gumbel_softmax`’s documentation is misleading](https://discuss.pytorch.org/t/function-gumbel-softmax-s-documentation-is-misleading/186157)](https://discuss.pytorch.org/t/function-gumbel-softmax-s-documentation-is-misleading/186157)
